### PR TITLE
Add is_zero to TypeVector and TypeTensor

### PR DESCRIPTION
**Pull request by aeslaughter**

### CODE REVIEW

**Comment by jwpeterson** [comment]
Error during fetch and branch... did you guys change up the recipes recently?
```
error: unknown option `shallow-submodules'
```

File: include/numerics/tensor_tools.h [hunk 1]
```diff
@@ -99,6 +99,10 @@ inline
 auto norm_sq(const VectorValue<T> & a) -> decltype(std::norm(T()))
 {return a.norm_sq();}
 
+template<typename T>
+inline
+auto is_zero(const T & a){ return a.is_zero();}
```
Comment (jwpeterson) [inline include/numerics/tensor_tools.h]: Return type should just be `bool`. IIRC automatic return type deduction is a C++14 thing. The other places where it is used in this class are for trailing return types.

**Comment by permcody** [comment]
Bet you wish you were here - we messed stuff up and we are at a "Full Stop" with CIVET right now! @#%@ is getting serious in here.

**Comment by pbauman** [comment]
Heh, for once it's not femputer being the trouble maker.

**Comment by permcody** [comment]
Yeah - we decided to throw in a few new options to "git clone" to keep our disk space usage down. Of course those options only work in newer version of git so all the clients just started dying on every job!

File: include/numerics/tensor_tools.h (hunk 1 above)
Comment (roystgnr) [reply include/numerics/tensor_tools.h]: You can do automatic return type deduction in C++11 but only with that awful redundant trailing arrow notation, not with just a single "auto".

**Comment by roystgnr** [comment]
Looks good after the C++11 compatibility is fixed IMHO.

File: include/numerics/tensor_tools.h (hunk 1 above)
Comment (aeslaughter) [reply include/numerics/tensor_tools.h]: Good, I don't like all the auto stuff anyway.

**Pull request merged**